src/passes/SimplifyLocals.cpp

```diff
@@ -577,6 +576,7 @@ struct SimplifyLocals
     this->replaceCurrent(newLocalSet);
     sinkables.clear();
     anotherCycle = true;
+    block->finalize();
```

Comment:
You mean, like [what the current code does](https://github.com/WebAssembly/binaryen/blob/f2ba91b6340a8c3099ab6a48700c717cbd33599e/src/passes/SimplifyLocals.cpp#L547-L549)? This function (`optimizeBlockReturn`) changes the block type, and this is the code where we recompute the new block type. We were able to compute the block type before by only checking the last argument's type, but now it is not possible because branches targeting this block might have other types, so we have to compute the least upper bound of all branch types + the last element type, which is why I called `block->finalize()` here. Can I do this without calling this again?